### PR TITLE
Adapted DataGrid to specs

### PR DESCRIPTION
**Pull request by greuelpirat**
I made several changes to approximate DataGrid to https://material.io/components/data-tables#specs

reworked-data-grid

visual changes:
* Row hover background (4 % light; 8 % dark) 134882d
* Fixed border and inner row lines issues (no more GridLines due to visual issues, They are now all Set via `DataGrid.BorderBrush` and `DataGrid.BorderThickness`) 7e1be8f 45876f8 98f8658
* CornerRadius 4dp b618329
* Long Column-Header trimmed and get AutoToolTip e96adbd
* Redesigned unused BottomThicknessConverter 88faa90 22f5fa8
* Updated Select-All-Cell to match rest of DataGrid 95ee20a
* Fixed multiple issues with Header- and Cell Padding to keep aligned 3828f98 7e91f6c 967246c b9466fa
* `SortDirectionIndicator` placed _outside_ so the do not move header-text and keep it aligned with cells 04fba45
* Removed explicit font size for MaterialDesignDataGridColumnHeader 1208d89
* Cell-Border _over_ cell to keep alignment 0e290b7

internal changes:
* Fixed warnings ef7b6c5
* Wrapped long lines 53861e1 112ebbd
* Updated DemoApp 29fe7d8 9e86394 59a671a 4b4336f

### CODE REVIEW

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.Light.xaml [hunk 1]
```diff
@@ -37,4 +37,6 @@
     <SolidColorBrush x:Key="MaterialDesignTextFieldBoxDisabledBackground" Color="#08000000" po:Freeze="True" /> <!-- 3% black -->
     <SolidColorBrush x:Key="MaterialDesignTextAreaBorder" Color="#BC000000" po:Freeze="True" /> <!-- 74% black -->
     <SolidColorBrush x:Key="MaterialDesignTextAreaInactiveBorder" Color="#0F000000" po:Freeze="True" /> <!-- 6% black -->
+
+    <SolidColorBrush x:Key="MaterialDesignDataGridRowHoverBackground" Color="#0A000000" po:Freeze="True" /> <!-- 4% black -->
```
[the remark this answers is not in the document]
Comment (greuelpirat) [reply MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.Light.xaml:41]: I added the definitions.

I also added tests with #1852 to ensure consistency.

**Comment by greuelpirat** [comment]
The open review issues are related to grid lines and I want to explain here the problem in details. There are some points, that I could fix using `GridLinesVisibility` and `HorizontalGridLinesBrush` instead of `BorderBrush` and `BorderThickness`. But the most annoying that I could not fix, was the bottom grid line when datagrid it self has a border. that looked really not good:

image

There is a doubled line at the datagrid bottom. The difference between both is more obvious when there is a horizontal scrollbar (one line above, one line below):

image

I found no way to get rid of the last grid line. so I tried to work around it. Because I wanted to make the datagrid "border-able" I tought the best way would be the make grid lines all by myself.

I also thought about something that the `GridLinesVisibility` is used from the styles to draw grid lines, but is not set on the inner DataGrid inside the template. This way the external usage would not be changed and the style would be applied well. But I could not make this approach work.

How would you solve this?

**Comment by greuelpirat** [comment]
* I updated GridLines to be controlled via `GridLinesVisibility` and `...GridLinesBrush`
* Set Default `BorderThickness` again to `0`

I see the bottom border https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/pull/1838#issuecomment-631867209 as a known issue. But since border is by default off, it should be no problem. I will continue to work on this in an other pull request.

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.DataGrid.xaml [hunk 2]
```diff
@@ -373,16 +410,15 @@
     <Style x:Key="MaterialDesignDataGrid" TargetType="{x:Type DataGrid}">
         <Setter Property="Background" Value="{DynamicResource MaterialDesignPaper}"/>
         <Setter Property="Foreground" Value="{DynamicResource MaterialDesignBody}"/>
-        <Setter Property="BorderBrush" Value="{DynamicResource MaterialDesignTextBoxBorder}"/>
-        <Setter Property="BorderThickness" Value="0"/>
+        <Setter Property="BorderBrush" Value="{DynamicResource MaterialDesignDivider}"/>
+        <Setter Property="BorderThickness" Value="1"/>
```
[the remark this answers is not in the document]
Comment (greuelpirat) [reply MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.DataGrid.xaml]: I set it back to `0`

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.DataGrid.xaml [hunk 3]
```diff
@@ -373,16 +410,15 @@
...
         <Setter Property="FontSize" Value="13"/>
-        <Setter Property="GridLinesVisibility" Value="Horizontal" />
+        <Setter Property="GridLinesVisibility" Value="None" />
```
[the remark this answers is not in the document]
Comment (greuelpirat) [reply MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.DataGrid.xaml]: Done

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.DataGrid.xaml [hunk 4]
```diff
@@ -329,6 +347,8 @@
         <Setter Property="Background" Value="Transparent"/>
         <Setter Property="SnapsToDevicePixels" Value="true"/>
         <Setter Property="Validation.ErrorTemplate" Value="{x:Null}"/>
+        <Setter Property="BorderBrush" Value="{Binding BorderBrush, RelativeSource={RelativeSource AncestorType=DataGrid}}" />
+        <Setter Property="BorderThickness" Value="{Binding BorderThickness, RelativeSource={RelativeSource AncestorType=DataGrid}, Converter={StaticResource TopThicknessConverter}}" />
```
[the remark this answers is not in the document]
Comment (greuelpirat) [reply MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.DataGrid.xaml]: I created a `GridLinesVisibilityBorderToThicknessConverter` to change that.

**Comment by Keboo** [comment]
Thanks for the explanation. After playing with it myself I think I agree that is likely the best we can do for now. Great work.

**Pull request merged**